operators/endpointmetrics/controllers/observabilityendpoint/observabilityaddon_controller.go

```diff
@@ -240,9 +240,13 @@ func (r *ObservabilityAddonReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 		deployer := deploying.NewDeployer(r.Client)
 		for _, res := range toDeploy {
-			if err := controllerutil.SetControllerReference(obsAddon, res, r.Scheme); err != nil {
-				log.Info("Failed to set controller reference", "resource", res.GetName())
-				globalRes = append(globalRes, res)
+			if !isHubMetricsCollector {
+				// For kind tests we need to deploy prometheus in hub but do not need to set controller
+				// reference as there is no observabilityaddon
```

Comment:
Do we **need not** or **can not**? Because if the case is that we don't need, but it's fine if we do it, we should just do it to simplify the code and avoid the `if`.

If we **can not**, then it would be worth to mention why in the comment fot future reference.

Reply:
There is no obsaddon for hub so we cannot. ill update it in the comments